example-get-started-cv/generate.sh

```diff
@@ -0,0 +1,198 @@
+#!/bin/bash
+# See https://dvc.org/doc/tutorials/get-started
+
+# Setup script env:
+#   e   Exit immediately if a command exits with a non-zero exit status.
+#   u   Treat unset variables as an error when substituting.
+#   x   Print commands and their arguments as they are executed.
+set -eux
+
+HERE="$( cd "$(dirname "$0")" ; pwd -P )"
+REPO_NAME="example-get-started-cv"
+REPO_PATH="$HERE/build/$REPO_NAME"
+PROD=${1:-false}
+
+if [ -d "$REPO_PATH" ]; then
+  echo "Repo $REPO_PATH already exists, please remove it first."
+  exit 1
+fi
+
+TOTAL_TAGS=15
+STEP_TIME=100000
+BEGIN_TIME=$(( $(date +%s) - ( ${TOTAL_TAGS} * ${STEP_TIME}) ))
+export TAG_TIME=${BEGIN_TIME}
+export GIT_AUTHOR_DATE="${TAG_TIME} +0000"
+export GIT_COMMITTER_DATE="${TAG_TIME} +0000"
+tick(){
+  export TAG_TIME=$(( ${TAG_TIME} + ${STEP_TIME} ))
+  export GIT_AUTHOR_DATE="${TAG_TIME} +0000"
+  export GIT_COMMITTER_DATE="${TAG_TIME} +0000"
+}
+
+export GIT_AUTHOR_NAME="Alex Kim"
+export GIT_AUTHOR_EMAIL="alex000kim@gmail.com"
+export GIT_COMMITTER_NAME="$GIT_AUTHOR_NAME"
+export GIT_COMMITTER_EMAIL="$GIT_AUTHOR_EMAIL"
+
+mkdir -p $REPO_PATH
+pushd $REPO_PATH
+
+virtualenv -p python3 .venv
+export VIRTUAL_ENV_DISABLE_PROMPT=true
+source .venv/bin/activate
+echo '.venv/' > .gitignore
+
+# Installing from main since we'd like to update repo before
+# the release
+pip install "git+https://github.com/iterative/dvc#egg=dvc[all]"
+
+git init
+cp $HERE/code/README.md .
+cp $HERE/code/.devcontainer.json .
+cp $HERE/code/.gitattributes .
+cp $HERE/code/requirements.txt .
+cp -r $HERE/code/.github .
+
+git add .
+tick
+git commit -m "Initialize Git repository"
+git tag -a "0-git-init" -m "Git initialized."
+
+dvc init
+tick
+git commit -m "Initialize DVC project"
+
+cp -r $HERE/code/data .
+git add data/.gitignore data/pool_data.dvc
+tick
+git commit -m "Add data"
+dvc pull
+
+# Remote active on this env only, for writing to HTTP redirect below.
+dvc remote add -d --local storage s3://dvc-public/remote/get-started-pools
+# Actual remote for generated project (read-only). Redirect of S3 bucket above.
+dvc remote add -d storage https://remote.dvc.org/get-started-pools
+git add .
+tick
+git commit -m "Configure default remote"
+dvc push
+
+
+cp -r $HERE/code/notebooks .
+pip install -r requirements.txt
+pip install jupyter
+jupyter nbconvert --execute 'notebooks/TrainSegModel.ipynb' --inplace
+git add .
+tick
+git commit -m "Add notebook using DVCLive"
+git tag -a "1-notebook-dvclive" -m "Notebook added."
+
+cp -r $HERE/code/src .
+cp $HERE/code/params.yaml .
+
+dvc stage add -n data_split \
+  -p base,data_split \
+  -d src/data_split.py -d data/pool_data \
+  -o data/train_data -o  data/test_data \
+  python src/data_split.py
+dvc repro
+git add .
+tick
+git commit -m "Create and run data split stage"
```

Comment:
Since this guide is not about stages (not much about them at least), let's simplify the number of tags, wdyt?

Reply:
Fine with me 👍